diff --git a/fesm2022/heddendorp-effect-platform-angular.mjs b/fesm2022/heddendorp-effect-platform-angular.mjs
index 2ba31531101118de712af69956ace27babf56239..8b96b5671f6664b18bd865d54fa27bd877549d9f 100644
--- a/fesm2022/heddendorp-effect-platform-angular.mjs
+++ b/fesm2022/heddendorp-effect-platform-angular.mjs
@@ -31,7 +31,10 @@ const resolveBody = (request) => {
         case 'Raw':
             return Effect.succeed(body.body);
         case 'Uint8Array':
-            return Effect.succeed(body.body);
+            if (typeof body.contentType === 'string' && body.contentType.includes('application/json')) {
+                return Effect.succeed(new TextDecoder().decode(body.body));
+            }
+            return Effect.succeed(body.body.buffer.slice(body.body.byteOffset, body.body.byteOffset + body.body.byteLength));
         case 'FormData':
             return Effect.succeed(body.formData);
         case 'Stream':
